deluge/ui/gtkui/piecesbar.py

```diff
@@ -73,7 +73,7 @@ def do_expose_event(self, event):
         # Restrict Cairo to the exposed area; avoid extra work
         self.__roundcorners_clipping()
 
-        if not self.__pieces and self.__num_pieces is not None:
+        if not self.__pieces and self.__num_pieces > 0:
```

Comment:
You shouldn't use numeric comparison where `None` is a known possible value. This should be a simple boolean comparison of self.__num_pieces as `0` and `None` return `False`.

What is the UI outcome if neither `__draw_pieces_completed()` or `__draw_pieces()` methods are called?